packages/jellyfish-api-core/src/category/masternode.ts

```diff
@@ -0,0 +1,146 @@
+   */
+  listMasternodes (pagination?: MasternodePagination, verbose?: boolean): Promise<MasternodesResult<MasternodeInfo>>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} verbose true
+   * @return {Promise<MasternodeResult>}
+   */
+  listMasternodes (pagination: MasternodePagination, verbose: true): Promise<MasternodesResult<MasternodeInfo>>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} verbose false.
+   * @return {Promise<MasternodeResul>}
+   */
+  listMasternodes (pagination: MasternodePagination, verbose: false): Promise<MasternodesResult<string>>
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} [verbose = true] Flag for verbose list. Only ids are returned when false.
+   * @return {Promise<MasternodeResult>}
+   */
+  async listMasternodes<T> (
+    pagination: MasternodePagination = {
+      including_start: true,
+      limit: 100
+    },
+    verbose: boolean = true
+  ): Promise<MasternodesResult<T>> {
+    return await this.client.call('listmasternodes', [pagination, verbose], 'number')
+  }
+
+  /**
+   * Returns information about a single masternode
+   *
+   * @param {string} masternodeId The masternode's id.
+   * @return {Promise<MasternodeResult>}
+   */
+  async getMasternode (masternodeId: string): Promise<MasternodeResult> {
+    return await this.client.call('getmasternode', [masternodeId], 'number')
+  }
+}
+
+export interface UTXO {
+  txid: string
+  vout: number
+}
+
+export interface CreateMasternodeOptions {
+  utxos: UTXO[]
+}
+
+export interface MasternodePagination {
+  start?: string
+  including_start?: boolean
+  limit?: number
+}
+
+export interface MasternodeInfo {
+  ownerAuthAddress: string
+  operatorAuthAddress: string
+  creationHeight: number
+  resignHeight: number
+  resignTx: string
+  banHeight: number
+  banTx: string
+  state: MasternodeState
+  mintedBlocks: number
+  ownerIsMine: boolean
+  operatorIsMine: boolean
+  localMasternode: boolean
+}
+
+export interface MasternodesResult<T> {
+  [id: string]: T
+}
+
+export interface MasternodeResult {
+  [id: string]: MasternodeInfo
+}
```

Comment:
```suggestion
export interface MasternodeResult<T> {
  [id: string]: T
}
```

Very potential confusing code, don't do this.